src/java.base/share/classes/java/lang/String.java

```diff
@@ -974,9 +974,12 @@ private static byte[] getBytesNoRepl1(String s, Charset cs) {
     private static byte[] encodeASCII(byte coder, byte[] val) {
         if (coder == LATIN1) {
             byte[] dst = Arrays.copyOf(val, val.length);
```

Comment:
Given the tweaks in https://git.openjdk.org/jdk/pull/12613 should this use `val.clone()` (would skip the length check)

```suggestion
            byte[] dst = val.clone();
```

Reply:
Yeah, probably makes sense. On that note I found that `val.clone()` underperform `Arrays.copyOf(val, val.length)` in C1 compiled code: https://bugs.openjdk.org/browse/JDK-8302850 - while this shouldn't affect peak performance it might be cause for a warmup regression.